behaviour/graql/reasoner/resolution/concept-inequality.feature

```diff
@@ -0,0 +1,398 @@
+      """
+      insert
+
+      $s1 isa state, has name 's1';
+      $s2 isa state, has name 's2';
+
+      (related-state: $s1) isa prior;
+      (related-state: $s1) isa achieved;
+      (related-state: $s2) isa achieved;
+      """
+#    When materialised keyspace is completed
+    Then for graql query
+      """
+      match (related-state: $s) isa holds; get;
+      """
+#    Then all answers are correct in reasoned keyspace
+    Then answer size in reasoned keyspace is: 1
+    Then for graql query
+      """
+      match
+        $s isa state, has name 's2';
+        (related-state: $s) isa holds;
+      get;
+      """
+#    Then all answers are correct in reasoned keyspace
+    Then answer size in reasoned keyspace is: 1
+#    Then materialised and reasoned keyspaces are the same size
+
+
+  Scenario: inferred binary relations can be filtered by concept inequality of their roleplayers
+#    When materialised keyspace is completed
+    Given for graql query
+      """
+      match
+        (ball1: $x, ball2: $y) isa selection;
+      get;
+      """
+#    Given all answers are correct in reasoned keyspace
+    # materialised: [ab, ba, bc, cb]
+    # inferred: [aa, ac, bb, ca, cc]
+    Given answer size in reasoned keyspace is: 9
+    Then for graql query
+      """
+      match
+        (ball1: $x, ball2: $y) isa selection;
+        $x != $y;
+      get;
+      """
+#    Then all answers are correct in reasoned keyspace
+    # materialised: [ab, ba, bc, cb]
+    # inferred: [ac, ca]
+    Then answer size in reasoned keyspace is: 6
+    # verify that the answer pairs to the previous query have distinct names within each pair
+    Then for graql query
+      """
+      match
+        (ball1: $x, ball2: $y) isa selection;
+        $x != $y;
+        $x has name $nx;
+        $y has name $ny;
+        $nx !== $ny;
+      get $x, $y;
+      """
+#    Then all answers are correct in reasoned keyspace
+    Then answer size in reasoned keyspace is: 6
+#    Then materialised and reasoned keyspaces are the same size
+
+
+  Scenario: inferred binary relations can be filtered by inequality to a specific concept
+#    When materialised keyspace is completed
+    Then for graql query
+      """
+      match
+        (ball1: $x, ball2: $y) isa selection;
+        $x != $y;
+        $y has name 'c';
+      get;
+      """
+#    Then all answers are correct in reasoned keyspace
+    Then answer size in reasoned keyspace is: 2
+    # verify answers are [ac, bc]
+    Then for graql query
+      """
+      match
+        (ball1: $x, ball2: $y) isa selection;
+        $x != $y;
+        $y has name 'c';
+        {$x has name 'a';} or {$x has name 'b';};
+      get;
+      """
+#    Then all answers are correct in reasoned keyspace
+    Then answer size in reasoned keyspace is: 2
+#    Then materialised and reasoned keyspaces are the same size
+
+  #
+  #   Tests a scenario in which the neq predicate binds free variables of two equivalent relations.
```

Comment:
We can actually put this kind of comment for the whole scenario as the scenario description, which wont then need the hashes if all goes well

Reply:
Updated, where applicable